src/main/java/org/tailormap/api/controller/admin/SolrAdminController.java

```diff
@@ -158,57 +171,110 @@ public ResponseEntity<?> pingSolr() {
       path = "${tailormap-api.admin.base-path}/index/{searchIndexId}",
       produces = MediaType.APPLICATION_JSON_VALUE)
   public ResponseEntity<?> index(@PathVariable Long searchIndexId) {
+    SearchIndex searchIndex = validateInputAndFindIndex(searchIndexId);
+
+    if (searchIndex.getStatus() == SearchIndex.Status.INDEXING) {
+      throw new ResponseStatusException(
+          HttpStatus.CONFLICT,
+          "Indexing already in progress, check tasks overview before retrying");
+    }
+
+    boolean createNewIndex =
+        (null == searchIndex.getLastIndexed()
+            || searchIndex.getStatus() == SearchIndex.Status.INITIAL);
+
+    boolean hasSchedule =
+        (null != searchIndex.getSchedule() && null != searchIndex.getSchedule().getUuid());
+
+    try {
+      if (hasSchedule) {
+        startScheduledJobIndexing(searchIndex);
+      } else {
+        startOneTimeJobIndexing(searchIndex);
+      }
+      searchIndexRepository.save(searchIndex);
+    } catch (UnsupportedOperationException
+        | IOException
+        | SolrServerException
+        | SolrException
+        | SchedulerException e) {
+      logger.error("Error indexing", e);
+      searchIndex.setStatus(SearchIndex.Status.ERROR);
+      searchIndexRepository.save(searchIndex);
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    }
+
+    if (createNewIndex) {
+      logger.info("Started creation of new index for search index {}", searchIndex.getName());
+      return ResponseEntity.status(HttpStatus.ACCEPTED).build();
+    } else {
+      logger.info("Started update of index for search index {}", searchIndex.getName());
+      return ResponseEntity.accepted().build();
```

Comment:
What's the difference between `ResponseEntity.status(HttpStatus.ACCEPTED)` and `ResponseEntity.accepted()` ?